python/pyspark/sql/connect/client/core.py

```diff
@@ -574,6 +576,8 @@ def __init__(
             the $USER environment. Defining the user ID as part of the connection string
             takes precedence.
         """
+        self.thread_local = threading.local()
```

Comment:
BTW, there is difference here. Python's threadlocal isn't inheritable (and there's no such implementation in Python). So we should somehow workaround this ...